src/Thorn_Calculus-Semantics-Synchronous.thy

```diff
@@ -145,6 +145,11 @@ proof -
     by force
 qed
 
+lemma repeated_receive_transition:
+  shows "A \<triangleright>\<^sup>\<infinity> x. \<P> x \<rightarrow>\<^sub>s\<lparr>A \<triangleright> \<star>\<^bsup>n\<^esup> X\<rparr> post_receive n X \<P> \<parallel> (A \<triangleright>\<^sup>\<infinity> x. \<P> x) \<guillemotleft> suffix n"
+  using receiving [where \<P> = "\<lambda>x. \<P> x \<parallel> A \<triangleright>\<^sup>\<infinity> x. \<P> x", unfolded post_receive_after_parallel]
+  by (subst repeated_receive_proper_def) force
```

Comment:
Using `force` here seems like overkill to me. A simple `simp` will do. In fact, even `unfold post_receive_def` would do.

Reply:
Oh, I just saw that I did the same in the code you replaced with an application of this lemma: https://github.com/input-output-hk/thorn-calculus/blob/9ebfc2045e0360ce713a7ec10e7ed6f12fc0c28e/src/Thorn_Calculus-Core_Bisimilarities.thy#L1545
Any ideas why I did that? :thinking:

Reply:
Actually I don't have any idea :grin: . But I think it would be nice to replace `force` with `unfold post_receive_def`. What do you think?

Reply:
Yes, sounds good.

Reply:
Fixed in 8732d7c4efca923f9ed3bdfc2329378246ddbc8d, please verify.

Reply:
Looks nice. :slightly_smiling_face: